src/Autofac/Core/IComponentRegistration.cs

```diff
@@ -92,9 +92,10 @@ public interface IComponentRegistration : IDisposable
         /// </summary>
         /// <param name="context">The context in which the instance will be activated.</param>
         /// <param name="parameters">Parameters for activation. These may be modified by the event handler.</param>
+        /// <param name="service">The service being resolved.</param>
         [SuppressMessage("Microsoft.Design", "CA1045:DoNotPassTypesByReference", MessageId = "1#", Justification = "The method may change the backing store of the parameter collection.")]
         [SuppressMessage("Microsoft.Design", "CA1030:UseEventsWhereAppropriate", Justification = "This is the method that would raise the event.")]
-        void RaisePreparing(IComponentContext context, ref IEnumerable<Parameter> parameters);
+        void RaisePreparing(IComponentContext context, ref IEnumerable<Parameter> parameters, Service service);
```

Comment:
Since we're making breaking changes, can I suggest that we move the ``service`` argument to before ``parameters``? Normally ref and out parameters would be last in the parameter list.

Reply:
Thanks for your suggestion. I have updated it. ;)